### ISSUE TITLE
[BUG] WSS  and CRLF method not working 

### ISSUE THREAD

**Issue by eliakimfelmorais** [OpenSIPS/opensips#2429]
```
version: opensips 3.1.1 (x86_64/linux)
flags: STATS: On, DISABLE_NAGLE, USE_MCAST, SHM_MMAP, PKG_MALLOC, Q_MALLOC, F_MALLOC, HP_MALLOC, DBG_MALLOC, FAST_LOCK-ADAPTIVE_WAIT
ADAPTIVE_WAIT_LOOPS=1024, MAX_RECV_BUFFER_SIZE 262144, MAX_LISTEN 16, MAX_URI_SIZE 1024, BUF_SIZE 65535
poll method support: poll, epoll, sigio_rt, select.
git revision: 424dc58
main.c compiled on 16:49:22 Mar  4 2021 with gcc 4.8.5
```

We are testing SIPJs library with OpenSIPS and wss layer and I got stucked with CRLF problem when I implment in UAC.

The UAC sends the ping to my OpenSIPS with WSS and the my WSS SIP Proxy does not respond with the pong message.

Looking on opensips.log I receive this message:

```
 16:18:59 [16216] INFO:core:parse_first_line: message too short: 0
 opensips: Mar  5 16:18:59 [16216] INFO:core:parse_first_line: bad message
 opensips: Mar  5 16:18:59 [16216] ERROR:core:parse_msg: message=<
 opensips: >
 opensips: Mar  5 16:18:59 [16216] ERROR:core:receive_msg: Unable to parse msg received from [189.90.58.138:55662]
 opensips: Mar  5 16:18:59 [16216] ERROR:proto_wss:ws_process: receive_msg failed

opensips: Mar  5 16:19:10 [16215] ERROR:nathelper:msg_send: send() to 189.90.58.138:51608 for proto wss/6 failed
opensips: Mar  5 16:19:10 [16215] ERROR:nathelper:nh_timer: sip msg_send failed
opensips: Mar  5 16:19:13 [16216] INFO:core:parse_first_line: message too short: 0
opensips: Mar  5 16:19:13 [16216] INFO:core:parse_first_line: bad message
opensips: Mar  5 16:19:13 [16216] ERROR:core:parse_msg: message=<
opensips: >
opensips: Mar  5 16:19:13 [16216] ERROR:core:receive_msg: Unable to parse msg received from [189.90.58.138:55662]
opensips: Mar  5 16:19:13 [16216] ERROR:proto_wss:ws_process: receive_msg failed

```
It seems the lib does not send the required information to CRLF (\r\n\r\n) behavior but I tested printing the data with LM_DBG and I got the string in my results.

I will attach the sip capture and you can you the this filter if you to look.

tcp.port==55662 && ip.src_host==189.90.58.138

**Comment by razvancrainea** [comment]
@eliakimfelmorais not sure why you consider this a bug, is there a requirement for the WS/WSS implementation to support this? In the initial design I don't remember seeing anything related to CRLF messages, only with standard WS/WSS ping & pong methods.
TBH, I haven't seen yet a webrtc implementation sending it yet, but if there are any (you're saying SIPjs does it - can it be disabled though?), we can swap this ticket into a feature request to support this.

PS: you forgot to attach the sip capture :).

**Comment by gmaruzz** [comment]
hello, Razvan,

yes, for a websocket endpoint there is a requirement to answer with a "PONG":

https://www.rfc-editor.org/rfc/rfc6455#section-5.5.2

cited in websocket as transport for SIP:

https://www.rfc-editor.org/rfc/rfc7118.html#section-6

**Comment by razvancrainea** [comment]
I agree, but that's already implemented and fully working. This ticket is asking for us to reply to a "TEXT"/"BIN" message containing a newline (CRLF). This is already done for the TCP and TLS protos, where there's no PING/PONG mechanism at the protocol level, but for WSS, we already have it (the one you pointed it out). That's why I don't see the reason for supporting "custom" ping messages, unless this is indeed widely spread across webrtc implementations. That's why I don't see it as a bug, but rather as a feature request.

**Comment by gmaruzz** [comment]
ooops, it seemed to me that it does not answer to CRLF ping sent via wss...
I will check again!
is there some special config needed, opensips side?

**Comment by razvancrainea** [comment]
Not clear what CRLF ping you're referring to - if it is the websocket ping (the ping detailed in the RFC as a special _opcode of 0x9_), that's supported. CRLF-s sent as text opcode are not.

**Comment by gmaruzz** [comment]
mmmmh, with 3.2-nightly, if I activate SIP.js 0.15.11 keepalive (which uses that CRLF ping) I get:

<pre>
Jul 28 16:22:00 lxc181 /usr/sbin/opensips[743]: ERROR:core:parse_msg: message=<#015#012#015#012>
Jul 28 16:22:00 lxc181 /usr/sbin/opensips[743]: ERROR:core:receive_msg: Unable to parse msg received from [5.170.45.166:43495]
Jul 28 16:22:00 lxc181 /usr/sbin/opensips[743]: ERROR:proto_wss:ws_process: receive_msg failed
Jul 28 16:22:10 lxc181 /usr/sbin/opensips[743]: ERROR:core:parse_msg: message=<#015#012#015#012>
Jul 28 16:22:10 lxc181 /usr/sbin/opensips[743]: ERROR:core:receive_msg: Unable to parse msg received from [5.170.45.166:43495]
Jul 28 16:22:10 lxc181 /usr/sbin/opensips[743]: ERROR:proto_wss:ws_process: receive_msg failed 
...
</pre>

**Comment by razvancrainea** [comment]
OK, so apparently SIP.js is the one doing this. I will turn this into a feature request.
BTW, is this behavior documented anywhere? is it expecting for a response - if so, what is it :)?

**Comment by gmaruzz** [comment]
no idea :)

they seems to use this as reference: https://datatracker.ietf.org/doc/html/rfc5626#section-4.4.1

btw, from this issue, it seems that at least asterisk works with it: https://github.com/onsip/SIP.js/pull/600
